pkg/operator/ceph/file/mds.go

```diff
@@ -141,11 +141,15 @@ func (c *cluster) start() error {
 		_, err := c.context.Clientset.ExtensionsV1beta1().Deployments(c.fs.Namespace).Create(d)
 		if err != nil {
 			if !errors.IsAlreadyExists(err) {
-				return fmt.Errorf("failed to create mds deployment: %+v", err)
+				return fmt.Errorf("failed to create mds deployment %s: %+v", mdsConfig.ResourceName, err)
+			}
+			logger.Infof("deployment for mds %s already exists. updating if needed", mdsConfig.ResourceName)
+			// TODO: need to prepare for upgrade here each time. Also, before a given deployment is
```

Comment:
I'd suggest for 0.9 we just need to document in the upgrade guide that the mds's will be updated automatically and not watch for disruption. We can improve this in the next release. For now we're just out of time.